Browser_IDE/CodeEditor.js

```diff
@@ -0,0 +1,95 @@
+class CodeEditor {
+    constructor(filename,codeblock_name,eleID,executionEnvironment, storedProject,filePath) {
+        this.filename = filename;
```

Comment:
A file path already contains the name of the file, so the `filename` parameter is redundant.

e.g. if `filePath` equals `/code/codeblock_init.js`, then we know that `filename` equals `codeblock_init.js`.

Reply:
do you mean the initCodePath equal to "codeblock_init.js"? I think it is different , as we don't know where the "mainLoopCodePath" and "initCodePath" come from, so we don't know what the value of it. but i am pretty sure it is not the something, as i used the filename as a filePath, but it throw the error, so this two variable is independent.

Reply:
These variables are defined in `projectInitializer.js`, but they are irrelevant to my point.
The file name can be derived from the file path; that does not mean that file path can be used directly as a file name.